upgrade_through_versions_test.py

```diff
@@ -15,160 +11,30 @@
-        # likewise for shorter versions of the form X.Y and X.Y-foo
-        # to accomplish this, check if "x.y.z-" (note the dash there) is contained within "x.y.z-foo", and if so declare x.y.z the higher version
-        # e.g. when comparing 3.0.0 and 3.0.0-rc1, consider 3.0.0 higher
-        # e.g. when comparing 3.3 and 3.3-beta1, consider 3.3 higher
-        if (len(self.semver.version) <= 3) or (len(other.semver.version) <= 3):
-            if self.semver.vstring + "-" in other.semver.vstring:
-                return 1
-            elif other.semver.vstring + "-" in self.semver.vstring:
-                return -1
-            elif other.semver.vstring == self.semver.vstring:
-                return 0
-
-        return cmp(self.semver, other.semver)
-
-
-def latest_tag_matching(ver_tuple):
-    """
-    Returns the latest tag matching a version tuple, such as (1, 2) to represent version 1.2
-    """
-    ver_str = make_ver_str(ver_tuple)
-
-    wrappers = []
-    # step through each tag found in the git repo
-    # check if the tag is a match for the base version provided in ver_tuple
-    # if it's a match add it to wrappers and when we complete this process give back the latest version found
-    for t in MAPPED_REFS['tags'].keys():
-        # let's short circuit if the tag we are checking matches the cassandra-x.y.z format, otherwise make another attempt for x.y.z-foo in case it's something line 1.2.3-tentative
-        match = re.match('^cassandra-({ver_str}\.\d+(-+\w+)*)$'.format(ver_str=ver_str), t) or re.match('^({ver_str}\.\d*(-+\w+)*)$'.format(ver_str=ver_str), t)
-        if match:
-            gsv = GitSemVer(t, match.group(1))
-            bisect.insort(wrappers, gsv)
-
-    if wrappers:
-        latest = wrappers.pop().git_ref
-        return latest
-
-    return None
+DEFAULT_PROTOCOL_VERSION = int(os.environ.get('DEFAULT_PROTOCOL_VERSION', 3))
 
 
-def make_ver_str(_tuple):
-    """
-    Takes a tuple like (1,2) and returns a string like '1.2'
-    """
-    return '{}.{}'.format(_tuple[0], _tuple[1])
-
-
-def make_branch_str(_tuple):
-    """
-    Takes a tuple like (1,2) and formats that version specifier as something
-    like 'cassandra-1.2' to match the branch naming convention
-    """
-
-    # special case trunk version to just return 'trunk'
-    if _tuple == TRUNK_VER:
-        return 'trunk'
-
-    return 'cassandra-{}.{}'.format(_tuple[0], _tuple[1])
-
-
-def sanitize_version(version):
+def sanitize_version(version, as_tuple=False):
     """
     Takes versions of the form cassandra-1.2, 2.0.10, or trunk.
     Returns just the version string 'X.Y.Z'
     """
-    if version.find('-') >= 0:
-        return LooseVersion(version.split('-')[1])
-    elif version == 'trunk':
-        return LooseVersion(make_ver_str(TRUNK_VER))
-    else:
-        return LooseVersion(version)
+    match = re.match('^.*(\d+\.+\d+\.*\d*).*$', unicode(version))
+    if match:
+        return LooseVersion(match.groups()[0])
```

Comment:
this will be appropriate in the future local context, since we won't be able to enforce any standard branch name onto folks in their local c* dev env. I'll whip up something to make it a bit more explicit though.